graphix/transpiler.py

```diff
@@ -945,6 +985,238 @@ def _rzz_command_opt(self, control_node, target_node, ancilla, angle):
+        seq.append(["M", ancilla[3], "XY", -0.0, [ancilla[2], ancilla[0]], []])
+        seq.append(["M", ancilla[5], "XY", -0.25, [ancilla[3], ancilla[1], ancilla[14], target_node], []])
+        seq.append(["M", control_node2, "XY", -0.25, [], []])
+        seq.append(["M", ancilla[6], "XY", -0.0, [ancilla[5], ancilla[2], ancilla[0]], []])
+        seq.append(["M", ancilla[9], "XY", -0.0, [control_node2, ancilla[0], ancilla[5], ancilla[2], ancilla[0]], []])
+        seq.append(["M", ancilla[7], "XY", -1.75, [ancilla[6], ancilla[3], ancilla[1], ancilla[14], target_node], []])
+        seq.append(["M", ancilla[10], "XY", -1.75, [ancilla[9], ancilla[14]], []])
+        seq.append(["M", ancilla[4], "XY", -0.25, [ancilla[14]], []])
+        seq.append(["M", ancilla[8], "XY", -0.0, [ancilla[7], ancilla[5], ancilla[2], ancilla[0]], []])
+        seq.append(
+            [
+                "M",
+                ancilla[11],
+                "XY",
+                -0.0,
+                [ancilla[10], control_node2, ancilla[0], ancilla[5], ancilla[2], ancilla[0]],
+                [],
+            ]
+        )
+        seq.append(
+            [
+                "M",
+                ancilla[12],
+                "XY",
+                -0.25,
+                [ancilla[8], ancilla[14], ancilla[6], ancilla[3], ancilla[1], ancilla[14], target_node],
+                [],
+            ]
+        )
+        seq.append(
+            [
+                "M",
+                ancilla[16],
+                "XY",
+                -0.0,
+                [
+                    ancilla[4],
+                    control_node1,
+                    ancilla[2],
+                    control_node2,
+                    ancilla[7],
+                    ancilla[10],
+                    ancilla[0],
+                    ancilla[0],
+                    ancilla[5],
+                    ancilla[2],
+                    ancilla[0],
+                    ancilla[5],
+                    ancilla[2],
+                    ancilla[0],
+                    control_node2,
+                    ancilla[0],
+                    ancilla[5],
+                    ancilla[2],
+                    ancilla[0],
+                ],
+                [],
+            ]
+        )
+        seq.append(["X", ancilla[17], [ancilla[14], ancilla[16]]])
+        seq.append(["X", ancilla[15], [ancilla[9], ancilla[11]]])
+        seq.append(["X", ancilla[13], [ancilla[0], ancilla[2], ancilla[5], ancilla[7], ancilla[12]]])
+        seq.append(["Z", ancilla[17], [ancilla[4], ancilla[5], ancilla[7], ancilla[10], control_node1]])
+        seq.append(["Z", ancilla[15], [control_node2, ancilla[2], ancilla[5], ancilla[10]]])
+        seq.append(["Z", ancilla[13], [ancilla[1], ancilla[3], ancilla[6], ancilla[8], target_node]])
+        return ancilla[17], ancilla[15], ancilla[13], seq
+
+    @classmethod
+    def _ccx_command_opt(self, control_node1, control_node2, target_node, ancilla):
```

Comment:
how did you get this opt version?

Reply:
I converted the quantum circuit to a ZX diagram and ran `id_simp` and `spider_simp` on the diagram. I tried other simplifications like `clifford_simp` and `full_reduce` but was not able to generate a pattern from the diagram.